packages/validator/test/unit/services/attestationDuties.test.ts

```diff
@@ -241,4 +242,137 @@ describe("AttestationDutiesService", () => {
 
     expect(api.validator.prepareBeaconCommitteeSubnet).toHaveBeenCalledOnce();
   });
+
+  describe("Reorg handling", () => {
+    const setupDuty = (slot: number): routes.validator.AttesterDuty => ({
+      slot,
+      committeeIndex: 1,
+      committeeLength: 120,
+      committeesAtSlot: 120,
+      validatorCommitteeIndex: 1,
+      validatorIndex: index,
+      pubkey: pubkeys[0],
+    });
+
+    async function setupTestEnvironment(duty: routes.validator.AttesterDuty, dependentRoot = "0xOLD_ROOT") {
+      api.validator.getAttesterDuties.mockResolvedValue(
+        mockApiResponse({
+          data: [duty],
+          meta: {dependentRoot, executionOptimistic: false},
+        })
+      );
+      api.validator.prepareBeaconCommitteeSubnet.mockResolvedValue(mockApiResponse({}));
+
+      const clock = new ClockMock();
+      const syncingStatusTracker = new SyncingStatusTracker(loggerVc, api, clock, null);
+
+      let onNewHeadCallback: ((headEvent: HeadEventData) => Promise<void>) | undefined;
+      chainHeadTracker.runOnNewHead.mockImplementation((callback) => {
+        onNewHeadCallback = callback;
+      });
+
+      const dutiesService = new AttestationDutiesService(
+        loggerVc,
+        api,
+        clock,
+        validatorStore,
+        chainHeadTracker,
+        syncingStatusTracker,
+        null
+      );
+
+      if (!onNewHeadCallback) throw new Error("onNewHeadCallback not initialized");
+      return {clock, dutiesService, onNewHeadCallback: onNewHeadCallback};
+    }
+
+    it("Should resubscribe to beacon subnets when current epoch dependent root changes", async () => {
+      const slot = 5;
+      const epoch = computeEpochAtSlot(slot);
+      const duty = setupDuty(slot);
+
+      const {clock, dutiesService, onNewHeadCallback} = await setupTestEnvironment(duty);
+      await clock.tickEpochFns(epoch, controller.signal);
+
+      expect(dutiesService["dutiesByIndexByEpoch"].get(epoch)).toBeDefined();
+      expect(api.validator.prepareBeaconCommitteeSubnet).toHaveBeenCalledTimes(1);
+
+      const reorgedDuty = {...duty, slot: slot + 1, committeeIndex: 3};
```

Comment:
the reorged duty would have the same `slot` as the original duty, so incrementing that by 1 is not right